test/unit/CodeGenerator/AttributeGeneratorTest.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OnMoon\OpenApiServerBundle\Test\Unit\Event\CodeGenerator;
+
+use OnMoon\OpenApiServerBundle\CodeGenerator\AttributeGenerator;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\DtoDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\GraphDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\OperationDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\PropertyDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\RequestDtoDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\ResponseDtoDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\SpecificationDefinition;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\Property;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\CodeGenerator\AttributeGenerator
+ */
+class AttributeGeneratorTest extends TestCase
+{
+    public function testSetAllAttributes(): void
+    {
+        $attributesGenerator = TestCase::createPartialMock(
+            AttributeGenerator::class,
+            ['requestPass', 'responsePass']
+        );
+
+        $graphMock          = TestCase::createMock(GraphDefinition::class);
+        $specDefinitionMock = TestCase::createMock(SpecificationDefinition::class);
+        $operationMock      = TestCase::createMock(OperationDefinition::class);
+        $requestMock        = TestCase::createMock(RequestDtoDefinition::class);
+        $responseMock       = TestCase::createMock(ResponseDtoDefinition::class);
+
+        $graphMock
+            ->expects(TestCase::once())
+            ->method('getSpecifications')
+            ->willReturn([$specDefinitionMock]);
+
+        $specDefinitionMock
+            ->expects(TestCase::once())
+            ->method('getOperations')
+            ->willReturn([$operationMock]);
+
+        $operationMock
+            ->expects(TestCase::once())
+            ->method('getRequest')
+            ->willReturn($requestMock);
+
+        $attributesGenerator
+            ->expects(TestCase::once())
+            ->method('requestPass')
+            ->with($requestMock);
+
+        $operationMock
+            ->expects(TestCase::once())
+            ->method('getResponses')
+            ->willReturn([$responseMock]);
+
+        $attributesGenerator
+            ->expects(TestCase::once())
+            ->method('responsePass')
+            ->with($responseMock);
+
+        $attributesGenerator->setAllAttributes($graphMock);
+    }
+
+    public function testRequestPass(): void
```

Comment:
We should also test other possible cases:
- when the property is required
- when the property has a default value
- when the property is nullable
- recursive calls when the property contains another object